source/production/F0.Minesweeper.Components/Logic/Game/GameLostUpdater.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using F0.Minesweeper.Components.Abstractions;
+using F0.Minesweeper.Components.Abstractions.Enums;
+using F0.Minesweeper.Components.Events;
+using Prism.Events;
+
+namespace F0.Minesweeper.Components.Logic.Game
+{
+	internal class GameLostUpdater : GameUpdater
+	{
+		private readonly IEventAggregator eventAggregator;
+
+		public GameLostUpdater(IEventAggregator eventAggregator)
+		{
+			this.eventAggregator = eventAggregator;
+		}
+
+		protected override Task OnUpdateAsync(IEnumerable<UncoverableCell> uncoverableCells, Minesweeper.Logic.Abstractions.Location clickedLocation)
+		{
+			foreach (var uncoverableCell in uncoverableCells)
+			{
+				var interaction = CellInteractionType.GameLost;
+
+				if (uncoverableCell.Cell.Location == clickedLocation)
+				{
+					interaction = CellInteractionType.LeftClick;
+				}
+
+				uncoverableCell.Cell.SetUncoveredStatus(interaction, uncoverableCell.IsMine, uncoverableCell.AdjacentMineCount);
+				uncoverableCell.Cell.DisableClick();
+			}
+
+			this.eventAggregator.GetEvent<GameFinishedEvent>().Publish("You lost the game!");
```

Comment:
so far - we don't use `this.` unless necessary

Reply:
yeah, you're right :) 
fixed!